src/main.cpp

```diff
@@ -115,748 +160,765 @@ int main ( int argc, char** argv )
+        auto result = QtAndroid::checkPermission ( QString ( "android.permission.RECORD_AUDIO" ) );
 
-    if ( result == QtAndroid::PermissionResult::Denied )
-    {
-        QtAndroid::PermissionResultMap resultHash = QtAndroid::requestPermissionsSync ( QStringList ( { "android.permission.RECORD_AUDIO" } ) );
-
-        if ( resultHash["android.permission.RECORD_AUDIO"] == QtAndroid::PermissionResult::Denied )
+        if ( result == QtAndroid::PermissionResult::Denied )
         {
-            return 0;
+            QtAndroid::PermissionResultMap resultHash = QtAndroid::requestPermissionsSync ( QStringList ( { "android.permission.RECORD_AUDIO" } ) );
+
+            if ( resultHash["android.permission.RECORD_AUDIO"] == QtAndroid::PermissionResult::Denied )
+            {
+                return 0;
+            }
         }
-    }
 #endif
 
 #ifdef _WIN32
-    // set application priority class -> high priority
-    SetPriorityClass ( GetCurrentProcess(), HIGH_PRIORITY_CLASS );
-
-    // For accessible support we need to add a plugin to qt. The plugin has to
-    // be located in the install directory of the software by the installer.
-    // Here, we set the path to our application path.
-    QDir ApplDir ( QApplication::applicationDirPath() );
-    pApp->addLibraryPath ( QString ( ApplDir.absolutePath() ) );
+        // set application priority class -> high priority
+        SetPriorityClass ( GetCurrentProcess(), HIGH_PRIORITY_CLASS );
+
+        // For accessible support we need to add a plugin to qt. The plugin has to
+        // be located in the install directory of the software by the installer.
+        // Here, we set the path to our application path.
+        if ( bUseGUI )
+        {
+            QDir ApplDir ( QApplication::applicationDirPath() );
+            pApplication->addLibraryPath ( QString ( ApplDir.absolutePath() ) );
+        }
+        else
+        {
+            QDir ApplDir ( QCoreApplication::applicationDirPath() );
+            pCoreApplication->addLibraryPath ( QString ( ApplDir.absolutePath() ) );
+        }
 #endif
 
 #if defined( Q_OS_MACX )
-    // On OSX we need to declare an activity to ensure the process doesn't get
-    // throttled by OS level Nap, Sleep, and Thread Priority systems.
-    CActivity activity;
+        // On OSX we need to declare an activity to ensure the process doesn't get
+        // throttled by OS level Nap, Sleep, and Thread Priority systems.
+        CActivity activity;
 
-    activity.BeginActivity();
+        activity.BeginActivity();
 #endif
 
-    // init resources
-    Q_INIT_RESOURCE ( resources );
+        // init resources
+        Q_INIT_RESOURCE ( resources );
 
 #ifndef SERVER_ONLY
-    // clang-format off
+        // clang-format off
 // TEST -> activate the following line to activate the test bench,
 //CTestbench Testbench ( "127.0.0.1", DEFAULT_PORT_NUMBER );
 // clang-format on
 #endif
 
-    CRpcServer* pRpcServer = nullptr;
```

Comment:
Smart pointer here - plus smart pointers for CServerRpc and CClientRpc - see my branch.